cmd/snap/cmd_quota.go

```diff
@@ -118,13 +166,62 @@ func parseQuotas(maxMemory string) (*client.QuotaValues, error) {
 		mem = value
 	}
 
+	if cpuMax != "" {
+		countValue, percentageValue, err := parseCpuQuota(cpuMax)
+		if err != nil {
+			return nil, err
+		}
+		if percentageValue > 100 || percentageValue <= 0 {
+			return nil, fmt.Errorf("cpu quota percentage must be between 1 and 100")
+		}
+
+		cpuCount = countValue
+		cpuPercentage = percentageValue
+	}
+
+	if cpuSet != "" {
+		cgv, err := getCGroupVersion()
+		if err != nil {
+			return nil, fmt.Errorf("cgroup v2 is required for --cpu-set: %v", err)
+		}
+		if cgv < 2 {
+			return nil, fmt.Errorf("cannot use --cpu-set with cgroup version %d", cgv)
+		}
+
+		cpuTokens := strings.Split(cpuSet, ",")
+		for i, cpuToken := range cpuTokens {
+			cpu, err := strconv.Atoi(cpuToken)
+			if err != nil {
+				return nil, fmt.Errorf("cannot parse value for --cpu-set at position %d", i)
+			}
+			if cpu < 0 || cpu >= runtime.NumCPU() {
+				return nil, fmt.Errorf("invalid cpu number %d in --cpu-set", cpu)
+			}
```

Comment:
I'm not sure we should be strict about the number of CPUs observed by go runtime at startup, as in theory CPUs can be brought online/offline dynamically on some systems, so maybe it's better to just assume that the operator knows what they are doing
```suggestion
			if cpu < 0 {
				return nil, fmt.Errorf("cannot use a negative CPU number", cpu)
			}
```

Reply:
Ah right, that makes a lot of sense.